### DISCUSSION TITLE
Yarn 2 / PnP / Zero-Install support?

### DISCUSSION THREAD

**Discussion by SoftCreatR** [parcel-bundler/parcel#6978] [Q&A] [answered]
I just stumbled across Yarn 2, which seems to be quite nice. However, I might be too stupid to understand, how it works exactly, or parcel is not compatible with the PnP / Zero-Install functionality (https://yarnpkg.com/features/zero-installs).

I've removed the `node_modules` directory and even the `yarn.lock` file, set `nodeLinker` to `pnp` in `.yarnrc.yml` and ran `yarn install`.

A new file `.pnp.cjs` has been created and all the other new Yarn magic happened. So far, so good.

But now, when running `yarn build` / `parcel build src/index.html --public-url '/'` (which works fine in Yarn 1 and even in Yarn 2 without PnP), I receive this error:

```
@parcel/core: Cannot find extended parcel config

/var/www/html/.parcelrc:2:13
  1 | {
> 2 |   "extends":"@parcel/config-default"
>   |             ^^^^^^^^^^^^^^^^^^^^^^^^ Cannot find module "@parcel/config-default"
  3 | }
  4 |
```

So I've temporarily removed the `.parcelrc` file and tried it again. This time, the build runs but it still fails, because:

```
@parcel/core: Failed to resolve '../node_modules/tachyons/css/tachyons.css' from './src/main.css'

  /var/www/html/src/main.css:1:9
  > 1 | @import '../node_modules/tachyons/css/tachyons.css';
  >   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

@parcel/resolver-default: Cannot load file '../node_modules/tachyons/css/tachyons.css' in './src'.
```

Which is correct, because with PnP there's no node_modules directory.

I guess, Parcel (v2) isn't fully compatible with Yarn v2 / PnP / Zero-Install yet?

**Comment by mischnic** [comment]
This works for me:

```
$ mkdir test && cd test
$ yarn set version berry
$ yarn init
$ yarn add parcel@nightly
$ echo "console.log('test');" > index.js
$ yarn parcel build index.js
```

**Reply by SoftCreatR** [reply]
I guess, `nodeLinker` is set to `node-modules`? Because that works for me too. Just not `pnp`.

**Reply by mischnic** [reply]
No, the resulting project does use pnp

**Reply by SoftCreatR** [reply]
Well, then I have no idea, why it's not working for me :(

**Comment by SoftCreatR** [comment, marked as answer]
Looks like I was able to resolve this issue by upgrading to the latest nightly (from 2.0.0-rc.0 to 2.0.0-nightly.837), which allowed me to use `npm:` instead of `../node_modules/.....`. PnP now works fine.
